bench/abc/optimized/solver.ll

```diff
@@ -2150,10 +2150,11 @@ clause_fetch.exit.i.i172.i.i:                     ; preds = %494, %.lr.ph86.i.i.
   br i1 %.not5581.i.i.i.i, label %.lr.ph.i.i174.i.i.preheader, label %.critedge57.loopexit.i.i.i.i
 
 .lr.ph.i.i174.i.i.preheader:                      ; preds = %518, %515, %504
+  %.ph = phi i32 [ 2, %504 ], [ 2, %515 ], [ %502, %518 ]
```

Comment:
Lots of regressions like this.

Reply:
maybe only do the fold if all arms can be folded to a constant to avoid this type of regression